### DISCUSSION TITLE
how to restore receiving address or pick one used before?

### DISCUSSION THREAD

**Discussion by pluto2021** [Chia-Network/chia-blockchain#5986] [Q&A]
If you click the "New Address" in the Wallet tab in the GUI, you will get a new receiving address, but the old one just disappeared without warning???!!! How to go back and choose the original address that has been setup for some other purpose?

**Comment by FSys77** [comment]
Very annoying indeed. I also would like to know this.

**Comment by DrakeFS** [comment]
Unless you saved it somewhere, I do not think it records a log of generated public addresses.

The old address is still valid though.

**Comment by tiny0little** [comment]
`$ chia keys show` shows it

`First wallet address: ...`

**Reply by pluto2021** [reply]
How to make that address appear in the GUI again? Shall we use the new address as receiving address? or shall we still use the hidden first wallet address as receiving address when setting up other services say on the exchange? Afraid of any bug that can lead to losing the reward.  Where is this new address stored? Can we manually change it back? Many thanks.

Regretted to have clicked that new address button, making things complicated.
